### DISCUSSION TITLE
How to indicate multicast support in SDP?

### DISCUSSION THREAD

**Discussion by cubisttriangle** [bluenviron/mediamtx#2977] [Questions] [answered]
### Question

In the context of RTSP, the SDP is typically included in the response to a DESCRIBE request and contains a c= line that specifies the connection information. If the server supports multicast, this line will include the multicast IP address and the Time to Live (TTL) value. Here's an example:

    c=IN IP4 224.1.0.0/32/1

In this example, IN represents the network type (Internet), IP4 represents the address type (IPv4), 224.1.0.0/32 is the multicast IP address and subnet mask, and 1 is the TTL value. This could let a client know that it should request a multicast session.

During my testing, I have only been able to observe c=IN IP4 0.0.0.0.

Is there a way to get mediamtx to modify this connection information to indicate that it supports multicast in the Session Description Protocol (SDP)?

Also, thank you so much for this project! It is very impressive!

**Comment by aler9** [comment, marked as answer]
Hello, yes, we did this to allow VLC to read multicast streams. It's enough to add `?vlcmulticast` to any URL. Documentation is here:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#vlc

**Reply by cubisttriangle** [reply]
That works! I had thought that was a vlc feature, not a mediamtx feature. Thanks for the help!

[a comment by an automated account is not shown]
